File: python/crypto_square.py
import re
import math


def cipher_text(plain_text):
    text = re.sub(r'[^a-zA-Z0-9]', '', plain_text).lower()
    row = round(len(text)**0.5)
    col = row
    if row * row < len(text):
        col = row + 1

    mat = [[' '] * col for _ in range(row)]

    i = 0
    j = 0
    for c in text:
        mat[i][j] = c
        j += 1
        if j == col:
            j = 0
            i += 1

    res = []
    for j in range(col):
        r = ''
        for i in range(row):
            r += mat[i][j]
        res.append(r)
    return ' '.join(res)

File: lua/crypto-square.lua
return {
    ciphertext = function(plaintext)
        plaintext = plaintext:gsub("[^%w]", ""):lower()
        if plaintext == "" then
            return ""
        end
        local root = (#plaintext) ^ 0.5
        local r = math.floor(root)
        local c = math.ceil(root)
        local fmt = string.format("%%-%ds", r * c)
        plaintext = string.format(fmt, plaintext)
        local t = {}
        local k = 0
        for i = 1, r do
            t[i] = {}
            for j = 1, c do
                t[i][j] = plaintext:sub(k, k)
                k = k + 1
            end
        end
        local o = {}
        for j = 1, c do
            table.insert(o, "")
            for i = 1, r do
                o[#o] = o[#o] .. t[i][j]
            end
        end
        return table.concat(o, " ")
    end,
}

